pkg/operator/rules.go

```diff
@@ -195,10 +195,24 @@ func ValidateRule(promRuleSpec monitoringv1.PrometheusRuleSpec) []error {
 		return []error{fmt.Errorf("the length of rendered Prometheus Rule is %d bytes which is above the maximum limit of %d bytes", promRuleSize, MaxConfigMapDataSize)}
 	}
 
-	_, errs := rulefmt.Parse(content, false)
+	_, errs := rulefmt.Parse(content, false, validationScheme)
 	return errs
 }
 
+func (prs *PrometheusRuleSelector) getValidationScheme() model.ValidationScheme {
+	// For Prometheus/PrometheusAgent 3.0.0+, use UTF-8 validation
+	// For ThanosRuler, always use legacy validation (ThanosRuler doesn't support UTF-8 yet)
```

Comment:
TODO: need to confirm this

Reply:
https://github.com/thanos-io/thanos/commit/0414eef64d6a3897ae4c09ac2fcb0aa1c1fb9cd1 landed in v0.38.0